src/transform/datastructures.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (C) 2021 NV Access Limited
+# This file may be used under the terms of the GNU General Public License, version 2 or later.
+# For more details see: https://www.gnu.org/licenses/gpl-2.0.html
+
+from dataclasses import dataclass
+from typing import Dict, Literal, NamedTuple, Optional, OrderedDict, Union
+
+AddonChannels = Literal["beta", "stable"]
```

Comment:
This is cool, I didn't know about it.

How does it compare to using an enum?

Reply:
I believe it's a Python3.8+ feature - I think you can use it in 3.7 with `typing_extensions`.
It's only useful for type hinting, as opposed to the other features of an Enum.

I think the only argument in favour of using it is that it is less rigid than an Enum.
If we wanted to, we could make this project accept any string for channel.

All that would need to change:
* accept any string for `channel`
* remove `generateAddonChannelDict`, instead create the channel keys as needed.

"beta" and "stable" are only used in tests and in these instances. If we wanted to use an `Enum` we could replace the test instances with a reference to the Enum member.

Reply:
So does this mean that if there isn't runtime validation for the values allowed?

In this case, the validation is already done by the json schema.

Reply:
Yes, exactly.
There is also run time validation in this instance, as only "beta" and "stable" channels are created in `generateAddonChannelDict`.
This could be changed to be handled by the schema only.

Reply:
Perhaps just add a comment to this literal to point out that the valid runtime values are defined by the json schema.

Reply:
Done as of b40a61d